include/deal.II/base/symbolic_function.h

```diff
@@ -0,0 +1,377 @@
+     */
+    static Tensor<1, dim, Differentiation::SD::Expression>
+    get_default_symbolic_position();
+
+    /**
+     * Get the actual arguments used for the coordinates in the symbolic
+     * function. This object does not include any user-defined arguments.
+     */
+    const Tensor<1, dim, Differentiation::SD::Expression> &
+    get_symbolic_position() const;
+
+    /**
+     * Get the actual symbolic time in use in this symbolic function.
+     */
+    const Differentiation::SD::Expression &
+    get_symbolic_time() const;
+
+    /**
+     * Get the actual symbolic expressions used in this symbolic function.
+     */
+    const std::vector<Differentiation::SD::Expression> &
+    get_symbolic_function_expressions() const;
+
+    /**
+     * Get the currently stored @p user_substitution_map.
+     */
+    const Differentiation::SD::types::substitution_map &
+    get_user_substitution_map() const;
+
+    /**
+     * Return a SymbolicFunction object that represents the time derivative of
+     * this function. The spatial argument, the symbolic time, and the currently
+     * stored user substitution map are forwarded to the new function.
+     */
+    SymbolicFunction<dim, RangeNumberType>
+    time_derivative() const;
+
+    /**
+     * Return the value of the function at the given point. Unless there is only
+     * one component (i.e., the function is scalar), you should state the
+     * component you want to have evaluated; it defaults to zero, i.e., the
+     * first component.
+     */
+    virtual RangeNumberType
+    value(const Point<dim> &p, const unsigned int component = 0) const override;
+
+    /**
+     * Return the gradient of the specified component of the function at the
+     * given point.
+     */
+    virtual Tensor<1, dim, RangeNumberType>
+    gradient(const Point<dim> & p,
+             const unsigned int component = 0) const override;
+
+    /**
+     * Compute the Laplacian of a given component at point <tt>p</tt>.
+     */
+    virtual RangeNumberType
+    laplacian(const Point<dim> & p,
+              const unsigned int component = 0) const override;
+
+    /**
+     * Compute the Hessian of a given component at point <tt>p</tt>, that is the
+     * gradient of the gradient of the function.
+     */
+    virtual SymmetricTensor<2, dim, RangeNumberType>
+    hessian(const Point<dim> & p,
+            const unsigned int component = 0) const override;
+
+    /**
+     * Print the stored arguments and function expression, as it would be
+     * evaluated when calling the method value().
+     */
+    template <typename StreamType>
+    StreamType &
+    print(StreamType &out) const;
+
+  private:
+    /**
+     * Return a substitution map that replaces the argument with the values of
+     * @p point, and the symbolic time with the value of this->get_time().
+     */
+    Differentiation::SD::types::substitution_map
+    substitute(const Point<dim> &point) const;
```

Comment:
if I understand correctly, then this function creates a substitution map that replaces coordinates by the values of `point`? If so, maybe call this function `create_substitution_map_for_point()`? That's because if the function is `const`, it can't actually *substitute* anything.